### PR TITLE
Windows: InvalidHandle requires constexpr

### PR DESCRIPTION
**Pull request by yadij**

### CODE REVIEW

File: src/base/File.h [hunk 1]
```diff
@@ -115,10 +115,10 @@ class File
     // Windows-specific HANDLE is needed because LockFileEx() does not take POSIX FDs.
 #if _SQUID_WINDOWS_ || _SQUID_MINGW_
     typedef HANDLE Handle;
-    static const Handle InvalidHandle;
+    static constexpr Handle InvalidHandle = INVALID_HANDLE_VALUE;
```
Comment (rousskov) [inline src/base/File.h:118]: AFAICT, theses changes must be propagated to the following InvalidHandle definition in src/base/File.cc (removing the latter, I guess):

```C++
#if _SQUID_WINDOWS_ || _SQUID_MINGW_
const HANDLE File::InvalidHandle = INVALID_HANDLE_VALUE;
#endif /* _SQUID_WINDOWS_ || _SQUID_MINGW_*/
```

**Review by rousskov — CHANGES_REQUESTED** [review]
```diff
- C++11: Windows InvalidHandle requires constexpr
+ Windows: InvalidHandle requires constexpr
```

I adjusted PR title to better reflect the nature/scope of these changes (and fix a technical bug). AFAICT, this PR is primarily about fixing or improving Windows build; The C++ standard version we use to do it is a minor detail that does not deserve such a prominent position in the PR title (and constexpr for static data members is a C++17, not C++11 [feature](https://en.cppreference.com/w/cpp/language/constexpr)).

Ideally, I would rephrase the title further to say that it fixes or improves Windows build, but I do not insist on that.

BTW, what build error or warning does this change fix on Windows? Please quote it (or its abbreviated version) in the PR description.

**Comment by yadij** [comment]
Turns out there are other issues with this that show up later. Closing for now.
